activation/validation.go

```diff
@@ -170,17 +170,9 @@ func (*Validator) VRFNonce(nodeId types.NodeID, commitmentAtxId types.ATXID, vrf
 	return nil
 }
 
-func (*Validator) InitialNIPostChallenge(challenge *types.NIPostChallenge, atxs atxProvider, goldenATXID types.ATXID) error {
-	if challenge.Sequence != 0 {
-		return fmt.Errorf("no prevATX declared, but sequence number not zero")
-	}
-
-	if challenge.InitialPost == nil {
-		return fmt.Errorf("no prevATX declared, but initial Post is not included in challenge")
-	}
-
+func (v *Validator) InitialNIPostChallenge(challenge *types.NIPostChallenge, atxs atxProvider, goldenATXID types.ATXID) error {
 	if challenge.CommitmentATX == nil {
-		return fmt.Errorf("no prevATX declared, but commitmentATX is missing")
+		v.log.Panic("commitment atx is nil")
```

Comment:
Why panic here? It opens a door to crash the node with a malicious ATX.

Reply:
to check programming error. it's already verified that in SyntacticallyValidate() that this cannot be nil. but anyway. changed to return error

Reply:
the idea was that all the nil were checked in the first stage of syntactic check. here it's doing checks on the deps. but not a big deal.